diff --git a/node_modules/@capacitor-community/sqlite/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsMigrate.java b/node_modules/@capacitor-community/sqlite/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsMigrate.java
index 351ee39..b5b8ade 100644
--- a/node_modules/@capacitor-community/sqlite/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsMigrate.java
+++ b/node_modules/@capacitor-community/sqlite/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/UtilsMigrate.java
@@ -74,7 +74,11 @@ public class UtilsMigrate {
     public String getFolder(Context context, String folderPath) throws Exception {
         String pathFiles = context.getFilesDir().getAbsolutePath();
         String pathDB = new File(context.getFilesDir().getParentFile(), "databases").getAbsolutePath();
-        if (!folderPath.equals("default")) {
+        if (folderPath.equals("default")) {
+            pathFiles = pathDB;
+        } else if (folderPath.equals("cache")) {
+            pathFiles = context.getCacheDir().getAbsolutePath();
+        } else {
             String[] arr = folderPath.split("/", 2);
             if (arr.length == 2) {
                 if (arr[0].equals("files")) {
@@ -85,8 +89,6 @@ public class UtilsMigrate {
                     throw new Exception("Folder " + folderPath + " not allowed");
                 }
             }
-        } else {
-            pathFiles = pathDB;
         }
         return pathFiles;
     }
