diff --git a/node_modules/capacitor-set-version/dist/common/utils-ios.js b/node_modules/capacitor-set-version/dist/common/utils-ios.js
index ac26188..e82cbbc 100644
--- a/node_modules/capacitor-set-version/dist/common/utils-ios.js
+++ b/node_modules/capacitor-set-version/dist/common/utils-ios.js
@@ -34,7 +34,7 @@ function setIOSVersionAndBuild(dir, version, build) {
     const projectFilePath = path.join(dir, exports.IOS_PROJECT_FILE_PATH);
     let file = openIOSProjectFile(projectFilePath);
     file = setIOSVersion(file, version);
-    file = setIOSBuild(file, build);
+    file = setIOSBuild(file, version);
     saveIOSProjectFile(projectFilePath, file);
 }
 exports.setIOSVersionAndBuild = setIOSVersionAndBuild;
@@ -43,7 +43,7 @@ function setIOSVersionAndBuildLegacy(dir, version, build) {
     let file = openInfoPlistFile(plistFilePath);
     const parsed = plist.parse(file);
     setIOSVersionLegacy(parsed, version);
-    setIOSBuildLegacy(parsed, build);
+    setIOSBuildLegacy(parsed, version);
     file = plist.build(parsed);
     saveInfoPlistFile(plistFilePath, file);
 }
